src/main/kotlin/com/example/moiza/domain/portfolio/service/ChangePublishService.kt

```diff
@@ -13,18 +14,24 @@ class ChangePublishService(
     private val portfolioRepository: PortfolioRepository,
 ) {
     @Transactional
-    fun execute(): Boolean {
+    fun execute(id: Long) {
         val user = userFacade.getCurrentUser()
-        val portfolio = portfolioRepository.findPortfolioByUser(user)
+        val portfolioList = portfolioRepository.findAllByUser(user)
+        val portfolio = portfolioRepository.findByIdOrNull(id)
             ?: throw PortfolioNotFoundException
 
-        if (portfolio.isPublished) {
-            user.updateUserStatus(UserStatus.PORTFOLIO_COMPLETED)
-        }
-        else {
+        if (!portfolio.isPublished) {
+            portfolioList?.forEach {
+                if (it.id == id) return@forEach;
+                if (it.isPublished) {
+                    it.changePublish()
+                }
+            }
```

Comment:
```suggestion
portfolioList.filter { it.id != id && it.isPublished }
    .forEach { it.changePublish() }

```
filter+forEach 방식으로 수정하는거 어떤가요?